pkg/database/token.go

```diff
@@ -271,7 +271,6 @@ func (db *Database) VerifyCodeAndIssueToken(request *IssueTokenRequest) (*Token,
 		// Check associated UserReport record if necessary.
 		if vc.UserReportID != nil {
 			providedNonce := base64.StdEncoding.EncodeToString(request.Nonce)
-			required := providedNonce != ""
```

Comment:
this was the bug - in the previous version it worked because it was all enforced in SQL. Now we need to check it after the query.